src/renderers/dom/shared/DOMProperty.js

```diff
@@ -197,33 +198,83 @@ var DOMProperty = {
   properties: {},
 
   /**
-   * Mapping from lowercase property names to the properly cased version, used
-   * to warn in the case of missing properties. Available only in __DEV__.
-   *
-   * autofocus is predefined, because adding it to the property whitelist
-   * causes unintended side effects.
-   *
-   * @type {Object}
+   * Some attributes are aliased for easier use within React. We don't
+   * allow direct use of these attributes. See DOMAttributeNames in
+   * HTMLPropertyConfig and SVGPropertyConfig.
    */
-  getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
+  aliases: {},
 
   /**
-   * All of the isCustomAttribute() functions that have been injected.
-   */
-  _isCustomAttributeFunctions: [],
-
-  /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (DOMProperty.isReservedProp(name)) {
+      return false;
+    }
+
+    if (value === null) {
+      return true;
+    }
+
+    var lowerCased = name.toLowerCase();
+
+    // Prevent aliases, and badly cased aliases like `class` or `cLASS`
+    // from showing up in the DOM
+    if (DOMProperty.aliases.hasOwnProperty(lowerCased)) {
+      return false;
+    }
+
+    var propertyInfo = DOMProperty.properties[name];
+
+    switch (typeof value) {
+      case 'boolean':
+        if (propertyInfo) {
+          return true;
+        }
+        var prefix = lowerCased.slice(0, 5);
+        return prefix === 'data-' || prefix === 'aria-';
+      case 'undefined':
+      case 'number':
+      case 'string':
         return true;
-      }
+      case 'object':
+        // Allow HAS_BOOLEAN_VALUE to coerce to true
+        if (propertyInfo && propertyInfo.hasBooleanValue) {
+          return true;
+        }
+        if (name === 'value') {
+          // Value is a special prop.
+          // Allow arrays or objects with custom toString.
+          return value.toString !== Object.prototype.toString;
```

Comment:
It's not ideal but we're not even sure we want to support it longer term. Probably not. But people (and we) already rely on it so maybe we can do it for a version before flipping it.